honeycomb-kernels/src/grisubal/model.rs

```diff
@@ -201,6 +203,37 @@ impl<T: CoordsFloat> From<Vtk> for Geometry2<T> {
     }
 }
 
+/// Check for orientation issue **per boundary**.
+///
+/// This function check for the most obvious orientation issue; given a boundary, are all segments making it up
+/// oriented consistently. If it is not the case, then there is at least one of:
+///
+/// - a vertex being the origin of two segment
+/// - a vertex being the end-point of two segment
+///
+/// This does not cover consistent orientation across distinct boundaries (e.g. a geometry with a hole in it).
+pub fn detect_orientation_issue<T: CoordsFloat>(
+    geometry: &Geometry2<T>,
+) -> Result<(), GrisubalError> {
+    let mut origins = HashSet::new();
+    let mut endpoints = HashSet::new();
+
+    for (orig, endp) in &geometry.segments {
+        if !origins.insert(orig) {
+            return Err(GrisubalError::InconsistentOrientation(
+                "two segments have the same vertex as their respective origin",
+            ));
```

Comment:
Can you add in the message the identifier that is detected as duplicated?

Reply:
I've done that in 47da67f, but it requires changing the type of the enum variant from `&'static str` to `String`